adoc/admin-monitoring-health-checks.adoc

```diff
@@ -0,0 +1,373 @@
+* Port http/10248: is only available on local host.
+
+You can perform an HTTP request to the endpoint to find out if the kubelet is healthy on that machine. The expected (healthy) HTTP response status code `200`.
+
+* Local Check
++
+If for example there is an agent running on each node, this agent can simply fetch the local healthz port:
++
+[source,bash]
+----
+curl -i http://localhost:10248/healthz
+----
+
+* Remote Check
++
+There are two ways to fetch endpoints remotely (metrics, healthz etc.). Both methods use HTTPS and a token.
++
+The first one is executed against the APIServer and mostly used with Prometheus and Kubernetes discovery https://prometheus.io/docs/prometheus/latest/configuration/configuration/#kubernetes_sd_config[kubernetes_sd_config], it allows automatic discovery of the nodes and avoids the task of defining monitoring for each node.
++
+The second method directly talks to kubelet can be used in more traditional monitoring where one must configure each node to be checked.
++
+** Configuration and Token retrieval
++
+Create a Service Account (monitoring) with a secondary Token (monitoring-secret-token) associated. The token will be used in HTTP requests to authenticate against the APIserver.
++
+This Service Account can only fetch information about nodes and pods. It is best practice to not use the default created token. Using a secondary token is also easier for management. Create a file [path]`kubelet.yaml` with the following as content.
++
+----
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: monitoring
+  namespace: kube-system
+secrets:
+- name: monitoring-secret-token
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: monitoring-secret-token
+  namespace: kube-system
+  annotations:
+    kubernetes.io/service-account.name: monitoring
+type: kubernetes.io/service-account-token
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: monitoring-clusterrole
+  namespace: kube-system
+rules:
+- apiGroups: [""]
+  resources:
+  - nodes/metrics
+  - nodes/proxy
+  - pods
+  verbs: ["get", "list"]
+- nonResourceURLs: ["/metrics", "/healthz", "/healthz/*"]
+  verbs: ["get"]
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRoleBinding
+metadata:
+  name: monitoring-clusterrole-binding
+  namespace: kube-system
+roleRef:
+  kind: ClusterRole
+  name: monitoring-clusterrole
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+- kind: ServiceAccount
+  name: monitoring
+  namespace: kube-system
+----
++
+Apply the yaml file
++
+[source,bash]
+----
+kubectl apply -f kubelet.yaml
+----
+Export the token to an environment variable:
++
+[source,bash]
+----
+TOKEN=$(kubectl -n kube-system get secrets monitoring-secret-token -o jsonpath='{.data.token}' | base64 -d)
+----
++
+This token can now be passed in headers in the form: "Authorization: Bearer $TOKEN"
```

Comment:
```suggestion
This token can now be passed with headers in the form: "Authorization: Bearer $TOKEN"
```

Reply:
Or `This token can now be passed in headers in the form`?

Reply:
I reworded it to: This token can now be passed through the `--header` argument as: "Authorization: Bearer $TOKEN".

Reply:
💯